Utilities/Development/list_extension_points.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+"""
+
+import os
+import json
+
+for d in os.walk('.'):
+  if d[0].startswith('./.git/'):
+    continue
+  if d[0].startswith('./sling/'):
+    continue
+  if '/target/' in d[0]:
+    continue
+  if '/node_modules/' in d[0]:
+    continue
```

Comment:
I'm not very good at python, but it seems to me that this whole set of conditions does nothing to affect the outcome. The last `if` will prune out all of these skipped directories anyway. I think a more efficient filter would be to ignore these directories **and** all their contents, with something like this:

```suggestion
  if d[0].startswith('./.git'):
    d[1][:] = []
    continue
  if d[0].startswith('./sling'):
    d[1][:] = []
    continue
  if d[0].endswith('/target'):
    d[1][:] = []
    continue
  if '/node_modules' in d[0]:
    d[1][:] = []
    continue
```